app/tests.py

```diff
@@ -976,6 +995,84 @@ def test_ajax_presto_query_endpoint(self):
         )
 
 
+class DownloadAttachment(TestCase):
+
+    def test_get_one_contract_attachment(self):
+        FAKE_ATTACHMENT_ID = 'FAKE_ATTACHMENT_ID_1'
+        FAKE_SF_QUERY_RESPONSES = (
+            {
+                'records': [
+                    {
+                        'Id': 'FAKE_ATTACHMENT_ID_1',
+                        'Name': 'FAKE_NAME_1',
+                        'ContentType': 'FAKE_CONTENT_TYPE_1',
+                    },
+                 ]
+            },
+        )
+        contract_id = ['FAKE_CONTRACT_ID_1']
+        sfq = _generate_fake_salesforce_query_instance()
+        with patch.object(sfq.sf, 'query', side_effect=FAKE_SF_QUERY_RESPONSES):
+            result = sfq.fetch_contract_attachments(contract_id)
+        for attachment in result:
+            self.assertEqual(attachment['salesforce_id'], FAKE_ATTACHMENT_ID)
+
+    def test_get_contract_attachments(self):
+        FAKE_ATTACHMENT_ID = ['FAKE_ATTACHMENT_ID_1', 'FAKE_ATTACHMENT_ID_2']
+        FAKE_SF_QUERY_RESPONSES = (
+            {
+                'records': [
+                    {
+                        'Id': 'FAKE_ATTACHMENT_ID_1',
+                        'Name': 'FAKE_NAME_1',
+                        'ContentType': 'FAKE_CONTENT_TYPE_1',
+                    },
+                    {
+                        'Id': 'FAKE_ATTACHMENT_ID_2',
+                        'Name': 'FAKE_NAME_2',
+                        'ContentType': 'FAKE_CONTENT_TYPE_2',
+                    },
+                 ]
+            },
+        )
+        contract_id = ['FAKE_CONTRACT_ID_1']
+        sfq = _generate_fake_salesforce_query_instance()
+        with patch.object(sfq.sf, 'query', side_effect=FAKE_SF_QUERY_RESPONSES):
+            result = sfq.fetch_contract_attachments(contract_id)
+        for attachment in result:
+            self.assertIn(attachment['salesforce_id'], FAKE_ATTACHMENT_ID)
+
+    def test_fetch_attachment(self):
+        FAKE_BODY = base64.b64encode(b"FAKE_CONTRACT_BODY")
+        FAKE_CONTENT_TYPE = "FAKE_CONTENT_TYPE"
+        FAKE_ATTACHMENT_ID = "FAKE_ATTACHMENT_ID"
+        sfq = _generate_fake_salesforce_query_instance()
+
+        with patch.object(requests, 'get', return_value=FAKE_BODY):
+            response = sfq.fetch_attachment(FAKE_ATTACHMENT_ID, FAKE_CONTENT_TYPE)
+
+        self.assertEqual(response, FAKE_BODY)
+
+    def test_download_attachment(self):
+        client = Client()
+        attachment = AttachmentFactory()
```

Comment:
We're not using this instance at all. 
Maybe we could add a test that creates attachments with this factory and then checks if they are listed in the template.